### PR TITLE
refactor: refactor UDP IR

### LINKED ISSUE #3353: Refactor XDS UDP Listener for UDPRoute
*Description*:
>Describe the issue.

Similar to https://github.com/envoyproxy/gateway/issues/1635, we need to make sure the XDS Listener name is based off the UDP Listener name to make sure the XDS listener does not flap when UDPRoutes churn.

Requires work similar to https://github.com/envoyproxy/gateway/pull/3271

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

### PR DESCRIPTION
**Pull request by aoledk**
**What this PR does / why we need it**:

Make sure the XDS Listener name is based off the UDP Listener name to make sure the XDS listener does not flap when UDPRoutes churn.

**Which issue(s) this PR fixes**:

Fixes #3353

### CODE REVIEW

**Review by shawnh2 — APPROVED** [review]
LGTM! Thanks for refactoring this!

File: internal/gatewayapi/helpers.go [hunk 1]
```diff
@@ -369,6 +365,10 @@ func irTCPRouteName(route RouteContext) string {
 	return fmt.Sprintf("%s/%s/%s", strings.ToLower(string(GetRouteType(route))), route.GetNamespace(), route.GetName())
 }
 
+func irUDPRouteName(route RouteContext) string {
+	return fmt.Sprintf("%s/%s/%s", strings.ToLower(string(GetRouteType(route))), route.GetNamespace(), route.GetName())
```
Comment (arkodg) [inline internal/gatewayapi/helpers.go:369]: can we just plug in `return irTCPRouteName(route)` here ? to reduce code duplication ?

Comment (aoledk) [reply internal/gatewayapi/helpers.go]: sure.

File: internal/xds/translator/translator.go [hunk 2]
```diff
@@ -529,29 +529,33 @@ func processUDPListenerXdsTranslation(tCtx *types.ResourceVersionTable, udpListe
...
+				errs = errors.Join(errs, err)
+				continue
+			}
+			if err := tCtx.AddXdsResource(resourcev3.ListenerType, xdsListener); err != nil {
+				// skip this listener if failed to add xds listener to the resource version table
+				errs = errors.Join(errs, err)
+				continue
+			}
+
+			// 1:1 between IR UDPListener and xDS Cluster
```
Comment (arkodg) [inline internal/xds/translator/translator.go:547]:
```suggestion
			// 1:1 between IR UDPRoute and xDS Cluster
```

**Comment by arkodg** [comment]
small typo, but LGTM overall, thanks for this refactor !
also heads up, trying to get https://github.com/envoyproxy/gateway/pull/3212 in because its been in the queue for a while
might cause this one to get rebased

**Review by arkodg — APPROVED** [review]

**Pull request merged**